### PR TITLE
Add ability to ban global functions

### LINKED ISSUE #327: Add ability to ban global functions
ie: `alert`

### PR DESCRIPTION
**Pull request by soniro**
Fixes #327.

This is my suggested solution. Please see the discussion in the issue about the syntax of global bans.

### CODE REVIEW

**Comment by JKillian** [comment]
Hi @soniro, thanks for the PR! Sorry for the delay in reviewing this. I'm happy with the syntax you proposed, just one question: would `["someFunc"]` ban `someObj.someFunc()` or only the standalone version of it?

File: src/rules/banRule.ts [hunk 1]
```diff
@@ -23,20 +23,18 @@ export class Rule extends Lint.Rules.AbstractRule {
     /* tslint:disable:object-literal-sort-keys */
     public static metadata: Lint.IRuleMetadata = {
         ruleName: "ban",
-        description: "Bans the use of specific functions.",
-        descriptionDetails: "At this time, there is no way to disable global methods with this rule.",
+        description: "Bans the use of specific functions or global methods.",
         optionsDescription: "A list of `['object', 'method']` pairs which ban `object.method()`.",
```
Comment (JKillian) [inline src/rules/banRule.ts]: Let's update this description as well?

File: test/rules/ban/test.ts.lint [hunk 2]
```diff
@@ -11,3 +11,13 @@ globals.getDocument().window.toString();
...
+describe("some text", () => {});
+xdescribe("some text", () => {});
+~~~~~~~~~    [function invocation disallowed: xdescribe]
+fdescribe("some text", () => {});
+~~~~~~~~~    [function invocation disallowed: fdescribe]
+it("some text", () => {});
+xit("some text", () => {});
+~~~          [function invocation disallowed: xit]
+fit("some text", () => {});
+~~~          [function invocation disallowed: fit]
```
Comment (JKillian) [inline test/rules/ban/test.ts.lint]: Could we add a test case for `someObject.fit()`?

**Comment by JKillian** [comment]
This PR looks good by the way! Looks like you'll need to merge master into it before we can merge though

**Comment by adidahiya** [comment]
ping @soniro?

**Comment by soniro** [comment]
Sorry for the delay.
Finally I added the suggestions from the review and merged the master.

Please let me know if you have problems to merge my pull request.

**Comment by chrismbarr** [comment]
Also related: #733

**Comment by JKillian** [comment]
Thanks @soniro!

**Comment by chrismbarr** [comment]
@soniro Does this still allow for custom ban messages like below?
```
["_", "each", "Don't rely on Underscore to perform loops. Use a 'for of' loop instead" ],
```

Would this allow for custom messages on the global function bans also?

**Pull request merged**